diff --git a/ash/webui/fyde_assistant_app_ui/resources/js/fyde_assistant_app.ts b/ash/webui/fyde_assistant_app_ui/resources/js/fyde_assistant_app.ts
index ad4182f7154cb..3bd9960f55fd4 100644
--- a/ash/webui/fyde_assistant_app_ui/resources/js/fyde_assistant_app.ts
+++ b/ash/webui/fyde_assistant_app_ui/resources/js/fyde_assistant_app.ts
@@ -28,6 +28,13 @@ interface MessageData {
   data?: any;
 }
 
+interface SystemColorInfo {
+  base: string;
+  shaded: string;
+  header: string;
+  primary: string;
+}
+
 const FYDE_ASSISTANT_APP_WEBVIEW_PARTITION = 'persist:fydeosAssistant';
 
 const FydeAssistantAppElementBase = PolymerElement;
@@ -64,6 +71,7 @@ export class FydeAssistantAppElement extends FydeAssistantAppElementBase {
   private debounceTimeout_: number;
   private isFromLauncher_: boolean;
   private isEventBinded_: boolean;
+  private systemColors_: SystemColorInfo;
 
   constructor() {
     super();
@@ -73,6 +81,7 @@ export class FydeAssistantAppElement extends FydeAssistantAppElementBase {
     this.debounceTimeout_ = 200;
     this.isFromLauncher_ = false;
     this.isEventBinded_ = false;
+    this.systemColors_ = { base: '', shaded: '', header: '', primary: '' };
   }
 
   override ready(): void {
@@ -114,21 +123,28 @@ export class FydeAssistantAppElement extends FydeAssistantAppElementBase {
     if (e) e.style.borderRadius = `${borderRadius}px`;
   }
 
-  append_value_to_url_(url: string, key: string) {
+  appendValueToUrl_(url: string, key: string) {
     if (loadTimeData.valueExists(key)) {
       return `${url}&${key}=${encodeURIComponent(loadTimeData.getString(key))}`;
     }
     return url;
   }
 
+  appendColorToUrl_(url: string, key: string) {
+    if (this.systemColors_[key as keyof SystemColorInfo]) {
+      return `${url}&${key}_color=${encodeURIComponent(this.systemColors_[key as keyof SystemColorInfo])}`;
+    }
+    return this.appendValueToUrl_(url, `${key}_color`);
+  }
+
   getUrl_() {
     let url = loadTimeData.getString('fydeosAssistantUrl');
     url = `${url}${document.location.search || '?'}`;
-    url = this.append_value_to_url_(url, 'user');
-    url = this.append_value_to_url_(url, 'base_color');
-    url = this.append_value_to_url_(url, 'shaded_color');
-    url = this.append_value_to_url_(url, 'header_color');
-    url = this.append_value_to_url_(url, 'primary_color');
+    url = this.appendValueToUrl_(url, 'user');
+    const keys = ['base', 'shaded', 'header', 'primary'];
+    for (let i = 0; i < keys.length; i++) {
+      url = this.appendColorToUrl_(url, keys[i]);
+    }
     return url;
   }
 
@@ -228,9 +244,9 @@ export class FydeAssistantAppElement extends FydeAssistantAppElementBase {
     this.sendMessage({ method: 'visibility-change', data: visible });
   }
 
-  onSystemColorChanged_(colors: {
-    base: string, shaded: string, header: string, primary: string,
-  }): void {
+  onSystemColorChanged_(colors: SystemColorInfo): void {
+    this.systemColors_ = colors;
+    console.log('this.systemColors_', this.systemColors_);
     this.sendMessage({ method: 'color-change', data: colors });
   }
 
